### PR TITLE
[3 of 3: CollapsingString] DCOS-8072: Implement CollapsingString component

### PR DESCRIPTION
**Pull request by jfurrow**
I implemented this everywhere I could think of that wouldn't be too performance intensive, as there are `resize` and `focus` handlers on `window` for each instance of the component.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @pierlo-upitup

**Comment by jfurrow** [comment]
And I broke all the page header tests... hold on for just a moment :)

**Comment by kennyt** [comment]
conflicts

**Comment by kennyt** [comment]
@jfurrow conflicts

**Comment by jfurrow** [comment]
@kennyt Flicts fixed, tests passing!

**Comment by rcorral** [comment]
retest this please

**Comment by jfurrow** [comment]
Additional conflicts fixed and tests still passing ;)

File: src/js/components/__tests__/ServiceDetail-test.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
+jest.dontMock('../CollapsingString');
 jest.dontMock('../PageHeader');
 jest.dontMock('../ServiceDetail');
 jest.dontMock('../ServiceDetailDebugTab');
 jest.dontMock('../ServiceDetailConfigurationTab');
 jest.dontMock('../ServiceDetailTaskTab');
 jest.dontMock('../ServiceInfo');
+jest.dontMock('../ServiceInfo');
```
Comment (kennyt) [inline src/js/components/__tests__/ServiceDetail-test.js]: don't need this line

**Comment by kennyt** [comment]
code looks good to me

**Pull request merged**